packet_spec.py

```diff
@@ -81,17 +118,84 @@ def parse_packet_header(header_bytes: bytes) -> PacketHeader:
+                    return TemperaturePacket(temperature=temperature, id=id, time_since_power=time)
+                case TelemetryPacketSubType.PRESSURE:
+                    pressure: int
+                    id: int
+                    time: int
+                    time, pressure, id = struct.unpack("<IIB", message_bytes)
+                    return PressurePacket(pressure=pressure, id=id, time_since_power=time)
+                case TelemetryPacketSubType.MASS:
+                    time: int
+                    mass: int
+                    id: int
+                    time, mass, id = struct.unpack("<IIB", message_bytes)
+                    return MassPacket(mass=mass, id=id, time_since_power=time)
+                case TelemetryPacketSubType.ARMING_STATE:
+                    time:int
+                    state:int
+                    time, state = struct.unpack("<IB", message_bytes)
+                    return ArmingStatePacket(state=ArmingState(state), time_since_power=time)
+                case TelemetryPacketSubType.ACT_STATE:
+                    time:int
+                    state:int
+                    id:int
+                    time, id, state = struct.unpack("<IBB", message_bytes)
+                    return ActuatorStatePacket(time_since_power=time, id=id, state=ActuatorState(state))
+                case TelemetryPacketSubType.WARNING:
+                    time:int
+                    type:int
+                    time, type = struct.unpack("<IB", message_bytes)
+                    return WarningPacket(type=Warning(type), time_since_power=time)
+
+#Demultiplexing the data and plotting
+def plot_point(plots, header, message):
     match header.type:
         case PacketType.CONTROL:
             pass
         case PacketType.TELEMETRY:
             match header.sub_type:
                 case TelemetryPacketSubType.TEMPERATURE:
-                    pass
+                    temperatureId:str = "t" + str(message.id)
+                    plots[temperatureId].points = np.append(plots[temperatureId].points, np.array([[message.time_since_power, message.temperature]]), axis=0)
+                    plots[temperatureId].data_line.setData(plots[temperatureId].points)
                 case TelemetryPacketSubType.PRESSURE:
-                    pass
+                    pressureId:str = "p" + str(message.id)
+                    plots[pressureId].points = np.append(plots[pressureId].points, np.array([[message.time_since_power, message.pressure]]), axis=0)
+                    plots[pressureId].data_line.setData(plots[pressureId].points)
                 case TelemetryPacketSubType.MASS:
-                    pass
+                    tankMass:str = "tank_mass"
+                    plots[tankMass].points = np.append(plots[tankMass].points, np.array([[message.time_since_power, message.mass]]), axis=0)
+                    plots[tankMass].data_line.setData(plots[tankMass].points)
```

Comment:
I think it might be a better idea if we keep all logic for updating the ui in `widget.py`. This will also avoid needing to pass in the plots and dataline as parameters as I'd imagine that'd slow things down a bit.